compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -1825,29 +1825,34 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
         if (scalaPrimitives.isPrimitive(sym)) {
           None
         } else if (isAbstractMethod(dd)) {
-          val body = if (scalaUsesImplClasses &&
-              sym.hasAnnotation(JavaDefaultMethodAnnotation)) {
-            /* For an interface method with @JavaDefaultMethod, make it a
-             * default method calling the impl class method.
-             */
-            val implClassSym = sym.owner.implClass
-            val implMethodSym = implClassSym.info.member(sym.name).suchThat { s =>
-              s.isMethod &&
-              s.tpe.params.size == sym.tpe.params.size + 1 &&
-              s.tpe.params.head.tpe =:= sym.owner.toTypeConstructor &&
-              s.tpe.params.tail.zip(sym.tpe.params).forall {
-                case (sParam, symParam) =>
-                  sParam.tpe =:= symParam.tpe
+          if (isNonNativeJSClass(currentClassSym)) {
```

Comment:
Could we avoid the nesting? (maybe by checking `isAbstractMethod` twice or putting it into a lazy val)?